src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -853,10 +897,78 @@ public SocketError ReceiveFrom(byte[] buffer, int offset, int count, ref SocketF
             }
             finally
             {
-                if (@event != null) @event.Dispose();
+                @event?.Dispose();
+            }
+        }
+
+        public unsafe SocketError ReceiveFrom(Span<byte> buffer, ref SocketFlags flags, byte[] socketAddress, ref int socketAddressLen, int timeout, out int bytesReceived)
```

Comment:
Do we need both this and the existing overload with (byte[],  int offset, int count)?  We can always go from byte[] -> Span, so why not just have this version and update the callers?